src/tests/string/test_string_functions.f90

```diff
@@ -52,6 +53,58 @@ subroutine test_reverse_string
 
     end subroutine test_reverse_string
 
+    subroutine test_slice_string
+        type(string_type) :: test_string
+        character(len=:), allocatable :: test_char
+        test_string = "abcdefghijklmnopqrstuvwxyz"
+        test_char = "abcdefghijklmnopqrstuvwxyz"
+
+        call check(slice(test_string, 2, 16, 3) == "behkn", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, first=15, stride=-1) == "onmlkjihgfedcba", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, last=22, stride=-1) == "zyxwv", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 7, 2) == "gfedcb", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, 7, 2, 1) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 2, 6, -1) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, stride=-1) == "zyxwvutsrqponmlkjihgfedcba", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, 7, 7, -4) == "g", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 7, 7, 3) == "g", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, 7, 7, 3) == "g", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 7, -10) == "gfedcba", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, 500, 22) == "zyxwv", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 50, 27) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, -20, -200) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, first=0, stride=-1) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, last=27, stride=-2) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, first=27, stride=2) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, -500, 500) == "abcdefghijklmnopqrstuvwxyz", &
+                    'function slice failed', warn=.false.)
+
+        test_string = ""
+        test_char = ""
+        call check(slice(test_string, 2, 16, 3) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 2, 16, 3) == "", &
+                    'function slice failed', warn=.false.)
```

Comment:
Try and come up with better descriptions for the aspect that each test is checking.

Reply:
I would highly recommend to keep the specific tests, because they should serve as example usage to help users understand the functionality.

Reply:
I agree with @awvwgk

Reply:
arranged test cases and added some descriptive comments so as to help a reader build the intuition for `slice`. Let me know if comments are good enough or not.

Reply:
You're headed in the right direction, now take the comments you added, and combine them into the description that gets printed if the check fails. I.e.

```
! Only one argument is given
          ! Valid
          call check(slice(test_string, first=10) == "jklmnopqrstuvwxyz") ! last=+inf
```
becomes
```
call check(slice(test_string, first=10) == "jklmnopqrstuvwxyz", "Only a valid argument for first is provided")
```

Reply:
The new versions are better. I'm good with it, but I suggest working on this area for future PRs. Think along the lines of "If the only thing a developer got was the output from failing tests, would they be able to understand what the code is supposed to do and then be able write a correct implementation?" Writing a good test suite is a skill and it takes practice, but it's worth it.